lib/mayaHydra/hydraExtensions/sceneIndex/registration.cpp

```diff
@@ -127,30 +132,62 @@ class PathInterfaceSceneIndex : public Fvp::PathInterfaceSceneIndexBase
         // 2) The second segment of the UFE path, with each UFE path component
         //    becoming an SdfPath component. If the last component is a number,
         //    then we are dealing with an instance selection.
-        SdfPath primPath = _sceneIndexPathPrefix;
         TF_AXIOM(appPath.nbSegments() == 2);
+        SdfPath primPath = _sceneIndexPathPrefix;
```

Comment:
Reworked UFE to Hydra prim selections logic to support selection of native instance prototypes subprims.